packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -71,5 +72,145 @@ def test_log_proxy(cluster):
+
+        task_stderr_response = cluster.get('{}?filter=STREAM:STDERR'.format(url))
+        check_response_ok(task_stderr_response, {})
+        task_stderr = task_stderr_response.content.decode('utf-8', 'ignore')
+        assert 'STDERR_LOG' in task_stderr, 'Missing `STDERR_LOG` in response. Got {}'.format(task_stderr)
+
+
+def get_task_url(cluster, task_name, stream=False):
+    """ The function returns a logging URL for a given task
+
+    :param cluster: cluster fixture
+    :param task_name: task name
+    :param stream: use range or stream endpoint
+    :return: url to get the logs for a task
+    """
+    state_response = cluster.get('/mesos/state')
+    check_response_ok(state_response, {})
+
+    framework_id = None
+    executor_id = None
+    slave_id = None
+    container_id = None
+
+    state_response_json = state_response.json()
+    assert 'frameworks' in state_response_json, 'Missing field `framework` in {}'.format(state_response_json)
+    assert isinstance(state_response_json['frameworks'], list), '`framework` must be list. Got {}'.format(
+        state_response_json)
+
+    for framework in state_response_json['frameworks']:
+        assert 'name' in framework, 'Missing field `name` in `frameworks`. Got {}'.format(state_response_json)
+        # search for marathon framework
+        if framework['name'] != 'marathon':
+            continue
+
+        assert 'tasks' in framework, 'Missing field `tasks`. Got {}'.format(state_response_json)
+        assert isinstance(framework['tasks'], list), '`tasks` must be list. Got {}'.format(state_response_json)
+        for task in framework['tasks']:
+            assert 'id' in task, 'Missing field `id` in task. Got {}'.format(state_response_json)
+            if not task['id'].startswith(task_name):
+                continue
+
+            assert 'framework_id' in task, 'Missing `framework_id` in task. Got {}'.format(state_response_json)
+            assert 'executor_id' in task, 'Missing `executor_id` in task. Got {}'.format(state_response_json)
+            assert 'id' in task, 'Missing `id` in task. Got {}'.format(state_response_json)
+            assert 'slave_id' in task, 'Missing `slave_id` in task. Got {}'.format(state_response_json)
+
+            framework_id = task['framework_id']
+            # if task['executor_id'] is empty, we should use task['id']
+            executor_id = task['executor_id']
+            if not executor_id:
+                executor_id = task['id']
+            slave_id = task['slave_id']
+
+            statuses = task.get('statuses')
+            assert isinstance(statuses, list), 'Invalid field `statuses`. Got {}'.format(state_response_json)
+            assert len(statuses) == 1, 'Must have only one status TASK_RUNNING. Got {}'.format(state_response_json)
+            status = statuses[0]
+            container_status = status.get('container_status')
```

Comment:
can we just make these assert status['container_status']? They way this is written now, I cannot tell if container_status isnt present or it is actually equal to None when it fails. If you do it with the single simple assert, a key error will tell you its not present, and the assertion error will tell you the value is empty. Also, in python '' will evaluate as boolean False like None so its best to use 'foo is not None' if an empty string is unacceptable